### PR TITLE
Change the api of DataParallel and Fleet

### PR DESCRIPTION
**Pull request by ForFishes**
### PR types

Others
### PR changes

APIs
### Describe

Change the interface usage of DP and Fleet.

### CODE REVIEW

File: python/paddle/distributed/fleet/base/fleet_base.py [hunk 1]
```diff
@@ -579,27 +597,25 @@ def distributed_optimizer(self, optimizer, strategy=None):
         """
         self.user_defined_optimizer = optimizer
 
-        if strategy == None:
-            strategy = DistributedStrategy()
+        if strategy is not None:
```
Comment (hutuxian) [inline python/paddle/distributed/fleet/base/fleet_base.py:601]: update the doc in distributed_optimizer?

Comment (ForFishes) [reply python/paddle/distributed/fleet/base/fleet_base.py:601]: done

File: python/paddle/distributed/fleet/base/fleet_base.py [hunk 2]
```diff
@@ -655,8 +672,9 @@ def forward(self, x):
         assert model is not None
         self.model = paddle.DataParallel(
             model,
-            group_size_limits=group_size_limits,
-            small_group_size=small_group_size)
+            comm_buffer_size=self._user_defined_strategy.fuse_grad_size_in_MB,
+            last_comm_buffer_size=self._user_defined_strategy.
```
Comment (gongweibao) [inline python/paddle/distributed/fleet/base/fleet_base.py:676]: Two lines?

Comment (ForFishes) [reply python/paddle/distributed/fleet/base/fleet_base.py:676]: 这个格式被自动调整了。太长了的一行python代码。

**Review by gongweibao — APPROVED** [review]
LGTM

**Review by hutuxian — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
lgtm

File: python/paddle/distributed/fleet/base/fleet_base.py [hunk 3]
```diff
@@ -87,12 +87,11 @@ class Fleet(object):
         .. code-block:: python
 
             import paddle.distributed.fleet as fleet
-
-            fleet.init()
-
             strategy = fleet.DistributedStrategy()
+            fleet.init(strategy)
```
Comment (wangxicoding) [inline python/paddle/distributed/fleet/base/fleet_base.py:91]: fleet.init(strategy=strategy)

Comment (ForFishes) [reply python/paddle/distributed/fleet/base/fleet_base.py:91]: fix in next pr.

**Pull request merged**